src/Lean/Server/Rpc/RequestHandling.lean

```diff
@@ -87,21 +98,17 @@ def registerBuiltinRpcProcedure (method : Name) paramType respType
   let proc := wrapRpcProcedure method paramType respType handler
   builtinRpcProcedures.modify fun ps => ps.insert method proc
 
-def registerRpcProcedure (method : Name) paramType respType
-    {paramLspType} [RpcEncoding paramType paramLspType] [FromJson paramLspType]
-    {respLspType} [RpcEncoding respType respLspType] [ToJson respLspType]
-    (handler : paramType → RequestM (RequestTask respType)) : CoreM Unit := do
+open Lean.Elab.Command in
+def registerRpcProcedure (method : Name) : CommandElabM Unit := do
```

Comment:
We could previously register a function `foo` under an arbitrary `method : Name`. I think this approach is actually better because fully resolved constant names are already guaranteed unique (right?), so that should avoid any possibility of conflict here.

Reply:
Yes, fully resolved constant names are unique.